packages/shorebird_cli/lib/src/executables/aot_tools.dart

```diff
@@ -82,10 +83,20 @@ class AotTools {
     return !_preLinkerFlutterRevisions.contains(flutterRevision);
   }
 
+  /// Runs the `aot-tools` executable with the given [command] and await for
+  /// its completion.
+  ///
+  /// If no [runCommand] is provided, [ShorebirdProcess.run] is used.
   Future<ShorebirdProcessResult> _exec(
     List<String> command, {
+    Future<ShorebirdProcessResult> Function(
+      String,
+      List<String>, {
+      String? workingDirectory,
+    })? runCommand,
```

Comment:
This should be a typedef

Reply:
If I make it as a private typedef I get a warning that since it is used in a single place, it should be inline.

I will keep this inline for now, and when we move this code up to shorebird process, I will transform it in a type def